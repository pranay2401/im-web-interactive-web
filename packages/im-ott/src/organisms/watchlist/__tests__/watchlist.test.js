import React from "react";
import { render, cleanup } from "@testing-library/react";
import Watchlist from "../Watchlist";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Watchlist should render correctly", () => {
  const tree = renderer.create(<Watchlist />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("This suit is to test the Watchlist Button component", () => {
  test("Snapshot of Add Remove Watchlist Button", () => {
    const { asFragment } = render(<Watchlist />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("add to watchlist", () => {
    const { getByTestId } = render(<Watchlist />);
    const ancestor = getByTestId("add-remove-watchlist-button");
    const plus = getByTestId("plus");

    expect(ancestor).toContainElement(plus);

    expect(getByTestId("add-remove-watchlist-label")).toHaveTextContent(
      "Watchlist"
    );
  });
});
