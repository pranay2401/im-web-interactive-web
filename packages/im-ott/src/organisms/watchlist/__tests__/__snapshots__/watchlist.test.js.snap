// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Watchlist/> should render correctly 1`] = `
<button
  className="ant-btn ant-btn-tertiary button undefined button tertiaryButton"
  data-testid="add-remove-watchlist-button"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="plus"
    className="anticon anticon-plus"
    data-testid="plus"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="plus"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <defs>
        <style />
      </defs>
      <path
        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
      />
      <path
        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
      />
    </svg>
  </span>
  <span
    className="label"
    data-testid="add-remove-watchlist-label"
  >
    Watchlist
  </span>
</button>
`;

exports[`This suit is to test the Watchlist Button component Snapshot of Add Remove Watchlist Button 1`] = `
<DocumentFragment>
  <button
    class="ant-btn ant-btn-tertiary button undefined button tertiaryButton"
    data-testid="add-remove-watchlist-button"
    type="button"
  >
    <span
      aria-label="plus"
      class="anticon anticon-plus"
      data-testid="plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs>
          <style />
        </defs>
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
    <span
      class="label"
      data-testid="add-remove-watchlist-label"
    >
      Watchlist
    </span>
  </button>
</DocumentFragment>
`;
